From: David Daney <ddaney@caviumnetworks.com>
Date: Wed, 10 Dec 2008 16:37:25 +0000 (-0800)
Subject: MIPS: Use EI/DI for MIPS R2.
X-Git-Url: http://www.linux-mips.org/git?p=linux.git;a=commitdiff_plain;h=72b2d2e290e2f18ec810da2089b01e3096d7d332

MIPS: Use EI/DI for MIPS R2.

For MIPS R2, use the EI and DI instructions to enable and disable
interrupts.

Signed-off-by: Tomaso Paoletti <tpaoletti@caviumnetworks.com>
Signed-off-by: David Daney <ddaney@caviumnetworks.com>
Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
(cherry picked from commit f15a6bca44fba50f62d0ebaa79a0db6bd72c2505)
---

diff --git a/include/asm-mips/asmmacro.h b/include/asm-mips/asmmacro.h
index 7a88175..6c8342a 100644
--- a/include/asm-mips/asmmacro.h
+++ b/include/asm-mips/asmmacro.h
@@ -35,6 +35,16 @@
 	mtc0	\reg, CP0_TCSTATUS
 	_ehb
 	.endm
+#elif defined(CONFIG_CPU_MIPSR2)
+	.macro	local_irq_enable reg=t0
+	ei
+	irq_enable_hazard
+	.endm
+
+	.macro	local_irq_disable reg=t0
+	di
+	irq_disable_hazard
+	.endm
 #else
 	.macro	local_irq_enable reg=t0
 	mfc0	\reg, CP0_STATUS
